gradle-plugin/test/proguard/gradle/ProguardCacheRelocateabilityIntegrationTest.groovy

```diff
@@ -0,0 +1,164 @@
+package proguard.gradle
+
+import org.apache.commons.io.FileUtils
+import org.gradle.testkit.runner.GradleRunner
+import org.junit.Rule
+import org.junit.rules.TemporaryFolder
+import spock.lang.Specification
+import spock.lang.Unroll
+
+class ProguardCacheRelocateabilityIntegrationTest extends Specification {
+    @Rule TemporaryFolder temporaryFolder
+    @Rule TemporaryFolder cacheFolder
+
+    @Unroll
+    def "proguard task can be relocated"() {
+        def cacheDir = cacheFolder.newFolder()
+        def originalDir = temporaryFolder.newFolder()
+        def fixture = new File(getClass().classLoader.getResource('spring-boot').path)
+        FileUtils.copyDirectory(fixture, originalDir)
+        writeBuildGradle(originalDir)
+        writeSettingsGradle(originalDir, cacheDir)
+
+        def relocatedDir = temporaryFolder.newFolder()
+        FileUtils.copyDirectory(fixture, relocatedDir)
+        writeBuildGradle(relocatedDir)
+        writeSettingsGradle(relocatedDir, cacheDir)
+
+        when:
+        def result = GradleRunner.create()
+            .forwardOutput()
+            .withArguments('proguard', '--build-cache')
+            .withPluginClasspath()
+            .withProjectDir(originalDir)
+            .build()
+
+        def result2 = GradleRunner.create()
+            .forwardOutput()
+            .withArguments('proguard', '--build-cache')
+            .withPluginClasspath()
+            .withProjectDir(relocatedDir)
+            .build()
+        then:
+        result.output =~ "SUCCESSFUL"
+        result2.output =~ "SUCCESSFUL"
+        result2.output =~ "Task :proguard FROM-CACHE"
+    }
+
+    def writeSettingsGradle(def projectDir, def cacheDir) {
+        new File(projectDir, "settings.gradle").text = """
+            rootProject.name = 'demo'
+            buildCache {
+                local(DirectoryBuildCache) {
+                    directory = "${cacheDir.absolutePath.replace(File.separatorChar, '/' as char)}"
+                }
+            }
+            """.stripIndent()
+    }
+
+    def writeBuildGradle(def projectDir) {
```

Comment:
There seems to be a lot of duplication here.  I didn't compare them to see what the differences are, but is it not feasible to do this with just some replacements and/or some added configuration?  I'm primarily concerned about changes in the example needing to be made in two places now.

Reply:
Thanks! This is done.